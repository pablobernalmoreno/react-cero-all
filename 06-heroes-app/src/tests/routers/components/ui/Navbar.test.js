import { mount } from "enzyme";
import { MemoryRouter, Router } from "react-router-dom";
import { AuthContext } from "../../../../auth/AuthContext";
import { Navbar } from "../../../../components/ui/Navbar";
import { types } from "../../../../types/types";

describe("Pruebas en el componente <Navbar />", () => {
  const historyMock = {
    push: jest.fn(),
    replace: jest.fn(),
    location: {},
    listen: jest.fn(),
    createHref: jest.fn(),
  };

  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: true,
      name: "Pablo",
    },
  };

  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter>
        <Router history={historyMock}>
          <Navbar />
        </Router>
      </MemoryRouter>
    </AuthContext.Provider>
  );

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("debe de mostrarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".text-info").text().trim()).toBe("Pablo");
  });

  test("debe de llamar el logout y usar history", () => {
    wrapper.find("button").simulate("click");
    expect(contextValue.dispatch).toHaveBeenCalledWith({ type: types.logout });

    expect(historyMock.replace).toHaveBeenCalledWith("/login");
  });
});
