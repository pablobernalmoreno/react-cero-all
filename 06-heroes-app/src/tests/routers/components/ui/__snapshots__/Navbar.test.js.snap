// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Navbar /> debe de mostrarse correctamente 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "3s3149",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "3s3149",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Router
      history={
        Object {
          "createHref": [MockFunction],
          "listen": [MockFunction],
          "location": Object {},
          "push": [MockFunction],
          "replace": [MockFunction],
        }
      }
    >
      <Navbar>
        <nav
          className="navbar navbar-expand-sm navbar-dark bg-dark"
        >
          <Link
            className="navbar-brand"
            to="/"
          >
            <LinkAnchor
              className="navbar-brand"
              navigate={[Function]}
            >
              <a
                className="navbar-brand"
                onClick={[Function]}
              >
                Asociaciones
              </a>
            </LinkAnchor>
          </Link>
          <div
            className="navbar-collapse"
          >
            <div
              className="navbar-nav"
            >
              <NavLink
                activeClassName="active"
                className="nav-item nav-link"
                exact={true}
                to="/marvel"
              >
                <Link
                  aria-current={null}
                  className="nav-item nav-link"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/marvel",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="nav-item nav-link"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="nav-item nav-link"
                      onClick={[Function]}
                    >
                      Marvel
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
              <NavLink
                activeClassName="active"
                className="nav-item nav-link"
                exact={true}
                to="/dc"
              >
                <Link
                  aria-current={null}
                  className="nav-item nav-link"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/dc",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="nav-item nav-link"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="nav-item nav-link"
                      onClick={[Function]}
                    >
                      DC
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
              <NavLink
                activeClassName="active"
                className="nav-item nav-link"
                exact={true}
                to="/search"
              >
                <Link
                  aria-current={null}
                  className="nav-item nav-link"
                  to={
                    Object {
                      "hash": "",
                      "pathname": "/search",
                      "search": "",
                      "state": null,
                    }
                  }
                >
                  <LinkAnchor
                    aria-current={null}
                    className="nav-item nav-link"
                    navigate={[Function]}
                  >
                    <a
                      aria-current={null}
                      className="nav-item nav-link"
                      onClick={[Function]}
                    >
                      Search
                    </a>
                  </LinkAnchor>
                </Link>
              </NavLink>
            </div>
          </div>
          <div
            className="navbar-collapse collapse w-100 order-3 dual-collapse2"
          >
            <ul
              className="navbar-nav ml-auto"
            >
              <span
                className="nav-item nav-link text-info"
              >
                Pablo
              </span>
              <button
                className="btn nav-item nav-link"
                onClick={[Function]}
              >
                Logout
              </button>
            </ul>
          </div>
        </nav>
      </Navbar>
    </Router>
  </Router>
</MemoryRouter>
`;
